money-api/src/main/java/com/comcast/money/api/SpanFactory.java

```diff
@@ -33,7 +33,7 @@
      * @return a child span with trace id and parent id from trace context header or a new root span if the
      * traceContextHeader is malformed.
      */
-    Span newSpanFromHeader(String childName, Function1<String,String> getHeader);
+    Span newSpanFromHeader(String childName, Function<String, String> getHeader);
```

Comment:
This is a binary breaking change but it enables both Java and Scala consumers to pass lambdas.

Reply:
You're going to have to forgive me but what would the issues be with breaking binary compatibility here? Just trying to think when that would come up for a user in normal usage

Reply:
I suspect that this API is probably rarely used directly.  It might be used if a consumer is using an HTTP framework that one of the existing Money projects doesn't support out of the box.  Currently calling this API from Java is kludgy given that Scala's `Function1[T, R]` type is not recognized as a functional interface, so Java users have to generate an anonymous class extending from `AbstractFunction<T, R>`.  This change enables both Java and Scala to call the function passing a lambda, since Scala 2.12+ supports functional interaces.